api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,180 @@
+ * would span implementations should go on the {@link ConditionService}.
+ */
+public class ConditionServiceImplTest extends BaseContextSensitiveTest {
+	
+	protected static final String CONDITION_XML = "org/openmrs/api/include/ConditionServiceImplTest-SetupCondition.xml";
+	
+	private ConditionService conditionService;
+	
+	private PatientService patientService;
+	
+	
+	@Before
+	public void setup (){
+		if(conditionService == null){
+			conditionService = Context.getConditionService();
+		}
+		if(patientService == null){
+			patientService = Context.getPatientService();
+		}
+		executeDataSet(CONDITION_XML);
+	}
+
+	/**
+	 * @see ConditionService#saveCondition(Condition) 
+	 */
+	@Test
+	public void saveCondition_shouldSaveNewCondition(){
+		Integer patientId = 2;
+		String uuid = "08002000-4469-12q3-551f-0339000c9a76";
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText();
+		Condition condition = new Condition();
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		condition.setUuid(uuid);
+		condition.setPatient(patientService.getPatient(patientId));
+		conditionService.saveCondition(condition);
+		Condition savedCondition = conditionService.getConditionByUuid(uuid);
+		Assert.assertNotNull(savedCondition.getConditionId());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		Integer conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(conditionId, condition.getConditionId());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getCondition_shouldFindConditionGivenValidId() {
+		int conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		String clinical_status = "INACTIVE";
+		String verification_status = "PROVISIONAL";
+		Condition condition = conditionService.getCondition(conditionId);
+		Assert.assertEquals(clinical_status, condition.getClinicalStatus().toString());
+		Assert.assertEquals(verification_status, condition.getVerificationStatus().toString());
+		Assert.assertEquals(uuid, condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getActiveConditions(Patient)
+	 */
+	@Test
+	public void getActiveConditions_shouldGetActiveConditions() {
+		int patientId = 2;
+		List<Condition> activeConditions = conditionService.getActiveConditions(patientService.getPatient(patientId));
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
```

Comment:
Same as here if you are using the variables only once

Reply:
Do the same for all other methods below

Reply:
Done